drivers/rtc/rtc_ll_stm32.c

```diff
@@ -76,46 +131,179 @@ struct rtc_stm32_config {
+					LL_RTC_AlarmTypeDef *ll_alarm_struct, uint16_t id)
 {
-	const struct rtc_stm32_config *cfg = dev->config;
+	/* RTC write protection is disabled & enabled again inside LL_RTC_ALMx_Init functions */
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	return LL_RTC_ALARM_Init(rtc, format, ll_alarm_struct);
+#else
+	if (id == RTC_STM32_ALRM_A) {
+		ll_alarm_struct->AlarmDateWeekDaySel = LL_RTC_ALMA_DATEWEEKDAYSEL_DATE;
+		return LL_RTC_ALMA_Init(rtc, format, ll_alarm_struct);
+	}
+	if (id == RTC_STM32_ALRM_B) {
+		ll_alarm_struct->AlarmDateWeekDaySel = LL_RTC_ALMB_DATEWEEKDAYSEL_DATE;
+		return LL_RTC_ALMB_Init(rtc, format, ll_alarm_struct);
+	}
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+	return 0;
+}
 
-	int err = 0;
+static inline void ll_func_clear_alarm_flag(RTC_TypeDef *rtc, uint16_t id)
+{
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	ARG_UNUSED(id);
 
-	uint32_t hour_format     = LL_RTC_GetHourFormat(RTC);
-	uint32_t sync_prescaler  = LL_RTC_GetSynchPrescaler(RTC);
-	uint32_t async_prescaler = LL_RTC_GetAsynchPrescaler(RTC);
+	LL_RTC_ClearFlag_ALR(rtc);
+#else
+	if (id == RTC_STM32_ALRM_A) {
+		LL_RTC_ClearFlag_ALRA(rtc);
+	} else {
+		LL_RTC_ClearFlag_ALRB(rtc);
+	}
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+}
 
-	LL_RTC_DisableWriteProtection(RTC);
+static inline uint32_t ll_func_is_active_alarm(RTC_TypeDef *rtc, uint16_t id)
+{
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	ARG_UNUSED(id);
+
+	return LL_RTC_IsActiveFlag_ALR(rtc);
+#else
+	if (id == RTC_STM32_ALRM_A) {
+		return LL_RTC_IsActiveFlag_ALRA(rtc);
+	}
+	if (id == RTC_STM32_ALRM_B) {
+		return LL_RTC_IsActiveFlag_ALRB(rtc);
+	}
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+	return 0;
+}
+
+static inline void ll_func_enable_interrupt_alarm(RTC_TypeDef *rtc, uint16_t id)
+{
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	ARG_UNUSED(id);
+
+	LL_RTC_EnableIT_ALR(rtc);
+#else
+	if (id == RTC_STM32_ALRM_A) {
+		LL_RTC_EnableIT_ALRA(rtc);
+	} else {
+		LL_RTC_EnableIT_ALRB(rtc);
+	}
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+}
+
+static inline void ll_func_disable_interrupt_alarm(RTC_TypeDef *rtc, uint16_t id)
+{
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	ARG_UNUSED(id);
+
+	LL_RTC_DisableIT_ALR(rtc);
+#else
+	if (id == RTC_STM32_ALRM_A) {
+		LL_RTC_DisableIT_ALRA(rtc);
+	} else {
+		LL_RTC_DisableIT_ALRB(rtc);
+	}
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+}
+
+static inline void ll_func_enable_alarm(RTC_TypeDef *rtc, uint16_t id)
+{
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	ARG_UNUSED(rtc);
+	ARG_UNUSED(id);
+#else
+	if (id == RTC_STM32_ALRM_A) {
+		LL_RTC_ALMA_Enable(rtc);
+	} else {
+		LL_RTC_ALMB_Enable(rtc);
+	}
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+}
+
+static inline void ll_func_disable_alarm(RTC_TypeDef *rtc, uint16_t id)
+{
+#ifdef CONFIG_SOC_SERIES_STM32F1X
+	ARG_UNUSED(rtc);
+	ARG_UNUSED(id);
+#else
+	if (id == RTC_STM32_ALRM_A) {
+		LL_RTC_ALMA_Disable(rtc);
+	} else {
+		LL_RTC_ALMB_Disable(rtc);
+	}
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+}
+#endif /* CONFIG_RTC_ALARM */
```

Comment:
Good but, please put these in a dedicated header file

Reply:
Done.